src/libasr/codegen/llvm_utils.cpp

```diff
@@ -2344,7 +2344,17 @@ namespace LCompilers {
         llvm::Type* el_type = std::get<2>(typecode2listtype[type_code]);
         resize_if_needed(list, current_end_point, current_capacity,
                          type_size, el_type, module);
-
+        llvm::Value* adjusted_pos = builder->CreateSelect(
+            builder->CreateICmpSGT(pos, current_end_point),
+            current_end_point,
+            pos
+        );
+        adjusted_pos = builder->CreateSelect(
+            builder->CreateICmpSGE(adjusted_pos, llvm::ConstantInt::get(context, llvm::APInt(32, 0))),
+            adjusted_pos,
+            llvm::ConstantInt::get(context, llvm::APInt(32, 0))
+        );
+        
```

Comment:
I guess so for this, we can use, [CreateMinimum](https://llvm.org/doxygen/classllvm_1_1IRBuilderBase.html#ada46861dd8cf60fa7e1ac61bbd385b49). Its just like, `min(len(list), position)`. If `position` will be greater than `len(list)` then automatically `position` will not be used and otherwise `position` will be used.

Reply:
Sorry I have not got very familiar with `llvm` , I tried - 
`llvm::Value* adjusted_pos = builder->CreateMinimum(pos, current_end_point);
`
But it leads to an error while running above test cases - 
```
attributes #0 = { argmemonly nounwind willreturn }
attributes #1 = { nounwind readnone speculatable willreturn }
code generation error: asr_to_llvm: module failed verification. Error:
Intrinsic has incorrect return type!
i32 (i32, i32)* @llvm.minimum.i32
```
Could you help me out here !

Reply:
I see. Why are we having, `adjusted_pos = builder->CreateSelect(
             builder->CreateICmpSGE(adjusted_pos, llvm::ConstantInt::get(context, llvm::APInt(32, 0))),
             adjusted_pos,
             llvm::ConstantInt::get(context, llvm::APInt(32, 0))
         );`?  For handling negative indices?

Reply:
@czgdp1807  Sorry I could not respond to this earlier, was busy with exams at university. 

Yes, that's correct. The code you mentioned is responsible for handling negative indices. Basically, it ensures new item is inserted at the beginning of the list if a negative index is provided.

Reply:
I remember a comment regarding not allowing negative indexing in the list ( allowed as of now ). You can check it [here](https://github.com/lcompilers/lpython/issues/1533#issuecomment-1445144248)
 
It thats the case I'll just remote the snippet checking for negative indices. Rest the pr looks good to me.